lib/game.rb

```diff
@@ -0,0 +1,49 @@
+class Game
+    include Interface
+
+    def initialize(player1, player2, board)
+        @player1 = player1
+        @player2 = player2
+        @board = board
+        @status = "initial"
+    end
+
+    def play
+        getInput('welcome')      
+        loop do   
+            restart if @status == "initial"
+            until game_finish? 
+                next_move(@player1) if !game_finish?
+                break if game_finish?
+                next_move(@player2) if !game_finish?
+            end 
+            break if !play_again?
+        end
+        finish 
+    end
+
+    private
+
+    def restart
```

Comment:
I like this method. It's very straightforward and you've crafted the respective instance methods of `Player` and `Board` nicely that you can simply call them out here. Only thing I think we can improve is to transfer this `restart` in a separate module (since this isn't per se a UI related method, we can have another module file that will house this method).

Reply:
ok, I completely remove this one, put the logic back to `main.rb`.